twoliter/embedded/rpm2img

```diff
@@ -196,7 +196,12 @@ if [[ -n "${CORE_KIT_VERSION}" && -n "${CORE_KIT_VENDOR}" ]]; then
   CORE_KIT_GIT_SHA="$(dnf --repofrompath \
     core-kit,file://"${CORE_KIT_PATH}" \
     --repo=core-kit repoquery \
-    --queryformat '%{Release}' 2>/dev/null | awk '/^1/{print $1}' | cut -d '.' -f 3)"
+    --queryformat '%{Buildtime} %{Release}' 2>/dev/null | \
+    sort -r -k 1,2 | \
+    head -n 1 | \
+    cut -d ' ' -f 3 | \
+    awk '/^1/{print $1}' | \
+    cut -d '.' -f 3)"
```

Comment:
`head` usage often leads to SIGPIPE since it doesn't keep reading from the pipe after a match, and we have `-o pipefail` set.

I would just pass it to `awk` after the initial sort:

```suggestion
    sort -r -k 1,2 | \
    awk -F '.' 'END {print $--NF}' )"
```

and maybe add a check that `CORE_KIT_GIT_SHA` is actually non-zero length.

Reply:
Nice, I really need to learn more `awk`. Only modification I made here was to undo the `-r` since we're grabbing the release from the last line instead of the first.